### PR TITLE
[golang] adding grpc port and env

### PR DESCRIPTION
**Pull request by tofluid**
Adding grpc support to service.yaml and deployment.yaml

### CODE REVIEW

File: charts/golang/templates/deployment.yaml [hunk 1]
```diff
@@ -127,6 +132,10 @@ spec:
           {{- if .Values.resources }}
           resources: {{- toYaml .Values.resources | nindent 12 }}
           {{- end }}
+          env:
+          {{- range .Values.env }}
+          - name: {{ .name }}
+            value: {{ .value }}
```
Comment (ellisio) [inline charts/golang/templates/deployment.yaml:138]: This isn't needed. We do this here:

https://github.com/fluidshare/helm-charts/blob/main/charts/golang/templates/deployment.yaml#L87-L89

**Review by ellisio — CHANGES_REQUESTED** [review]

File: charts/golang/templates/deployment.yaml (hunk 1 above)
Comment (tofluid) [reply charts/golang/templates/deployment.yaml]: removed

File: charts/golang/templates/service.yaml [hunk 2]
```diff
@@ -33,4 +33,14 @@ spec:
       {{- else if eq .Values.service.type "ClusterIP" }}
       nodePort: null
       {{- end }}
+    {{- if .Values.service.grpc }}
+    - name: grpc
+      port: {{ .Values.service.grpc.port }}
+      targetPort: grpc
+      {{- if and (or (eq .Values.service.type "NodePort") (eq .Values.service.type "LoadBalancer")) (not (empty .Values.service.nodePort)) }}
+      nodePort: {{ .Values.service.grpc.nodePort  }}
```
Comment (ellisio) [inline charts/golang/templates/service.yaml:41]: @tofluid Couple things here:

1. It looks like you're missing the `service.grpc` block within `values.yaml` to provide a default configuration, with documentation on what that block is for.
2. The last condition on line 40 needs to be `empty .Values.service.grpc.nodePort`.

**Review by ellisio — CHANGES_REQUESTED** [review]

File: charts/golang/templates/deployment.yaml [hunk 3]
```diff
@@ -84,6 +84,12 @@ spec:
                   name: {{ .Release.Name }}-redis
                   key: redis-password
             {{- end }}
+            - name: applicationPort
+              value: {{ quote .Values.applicationPort }}
+            {{- if .Values.service.grpc }}
+            - name: applicationGrpcPort
+              value: {{ quote .Values.applicationGrpcPort }}
+            {{- end }}
```
Comment (ellisio) [inline charts/golang/templates/deployment.yaml:92]: This block seems a little confusing. If GRPC is enabled, you're looking for a root value. Can this be refactored to look for `.Values.service.grpc.port`? That way you can put a default value in the `service.grpc` config along with the `nodePort`.

**Review by ellisio — CHANGES_REQUESTED** [review]

File: charts/golang/templates/deployment.yaml (hunk 3 above)
Comment (tofluid) [reply charts/golang/templates/deployment.yaml:92]: i had done that originally then decided that i wanted to try to match how http was done. I'll revert

File: charts/golang/templates/service.yaml (hunk 2 above)
Comment (tofluid) [reply charts/golang/templates/service.yaml:41]: 1. Added to values.yaml
2. changed

File: charts/golang/templates/deployment.yaml [hunk 4]
```diff
@@ -84,6 +84,12 @@ spec:
                   name: {{ .Release.Name }}-redis
                   key: redis-password
             {{- end }}
+            - name: applicationPort
+              value: {{ quote .Values.applicationPort }}
+            {{- if .Values.service.grpc }}
```
Comment (ellisio) [inline charts/golang/templates/deployment.yaml:89]: Lets do a check here for `.Values.service.grpc.port`.

File: charts/golang/templates/deployment.yaml [hunk 5]
```diff
@@ -84,6 +84,12 @@ spec:
...
+            - name: applicationGrpcPort
+              value: {{ quote .Values.service.grpc.applicationPort }}
```
Comment (ellisio) [inline charts/golang/templates/deployment.yaml:91]: Reference `.Values.service.grpc.port` instead.

File: charts/golang/templates/deployment.yaml [hunk 6]
```diff
@@ -99,6 +105,10 @@ spec:
           ports:
             - name: http
               containerPort: {{ .Values.applicationPort }}
+          {{- if .Values.service.grpc }}
```
Comment (ellisio) [inline charts/golang/templates/deployment.yaml:108]: Lets do a check here for `.Values.service.grpc.port`.

File: charts/golang/templates/deployment.yaml [hunk 7]
```diff
@@ -99,6 +105,10 @@ spec:
...
+            - name: grpc
+              containerPort: {{ .Values.service.grpc.applicationPort }}
```
Comment (ellisio) [inline charts/golang/templates/deployment.yaml:110]: Reference `.Values.service.grpc.port` instead.

File: charts/golang/templates/service.yaml [hunk 8]
```diff
@@ -33,4 +33,14 @@ spec:
       {{- else if eq .Values.service.type "ClusterIP" }}
       nodePort: null
       {{- end }}
+    {{- if .Values.service.grpc }}
```
Comment (ellisio) [inline charts/golang/templates/service.yaml:36]: Lets do a check here for `.Values.service.grpc.port`.

File: charts/golang/values.yaml [hunk 9]
```diff
@@ -173,6 +173,20 @@ service:
   ## HTTP Port
   ##
   port: 80
+
```
Comment (ellisio) [inline charts/golang/values.yaml:176]: If you're using an editor that supports EditorConfig, make sure you have that plugin installed. I think you can do this in any IntelliJ IDE.

File: charts/golang/values.yaml [hunk 10]
```diff
@@ -173,6 +173,20 @@ service:
   ## HTTP Port
   ##
   port: 80
+  
+  ## For defining a GRPC Port
+  ##
+  # grpc:
```
Comment (ellisio) [inline charts/golang/values.yaml:179]: Leave this line uncommented.

File: charts/golang/values.yaml [hunk 11]
```diff
@@ -173,6 +173,20 @@ service:
...
+    ## Specify the port where your application's grpc will be listening from externally
+    ##
+    # applicationPort: 8090
```
Comment (ellisio) [inline charts/golang/values.yaml:182]: You can remove this comment and line.

**Review by ellisio — CHANGES_REQUESTED** [review]

File: charts/golang/templates/deployment.yaml (hunk 4 above)
Comment (tofluid) [reply charts/golang/templates/deployment.yaml:89]: in this case i'll check for `applicationPort` as that's what's used in this block

File: charts/golang/values.yaml (hunk 10 above)
Comment (tofluid) [reply charts/golang/values.yaml:179]: done

File: charts/golang/templates/deployment.yaml (hunk 5 above)
Comment (tofluid) [reply charts/golang/templates/deployment.yaml]: done

File: charts/golang/templates/deployment.yaml (hunk 6 above)
Comment (tofluid) [reply charts/golang/templates/deployment.yaml]: done

File: charts/golang/templates/service.yaml (hunk 8 above)
Comment (tofluid) [reply charts/golang/templates/service.yaml]: done

File: charts/golang/values.yaml (hunk 9 above)
Comment (tofluid) [reply charts/golang/values.yaml]: done

File: charts/golang/values.yaml (hunk 11 above)
Comment (tofluid) [reply charts/golang/values.yaml]: done

**Review by ellisio — APPROVED** [review]

**Pull request merged**